db/migrate/20150511185213_create_users.rb

```diff
@@ -0,0 +1,14 @@
+class CreateUsers < ActiveRecord::Migration
+  def change
+    create_table :users do |t|
+      t.string :provider,    null: false
+      t.string :uid,         null: false
+      t.string :nickname,    null: false
```

Comment:
Does `nickname` represent the user's GitHub username? If so, I'd recommend using `login` for consistency with the GitHub API: https://developer.github.com/v3/users/#get-a-single-user